authzed/api/v1/permission_service.proto

```diff
@@ -172,16 +172,36 @@ message SubjectFilter {
 // relationships within the system.
 message ReadRelationshipsRequest {
   Consistency consistency = 1;
+
+  // relationship_filter defines the filter to be applied to the relationships
+  // to be returned.
   RelationshipFilter relationship_filter = 2
       [ (validate.rules).message.required = true ];
+
+  // optional_limit, if non-zero, specifies the limit on the number of relationships to return
+  // before the stream is closed on the server side. By default, the stream will continue
+  // resolving relationships until exhausted or the stream is closed due to the client or a
+  // network issue.
+  uint32 optional_limit = 3 [(validate.rules).uint32 = {gte:0, lte: 1000}];
```

Comment:
We shouldn't validate this at the API level IMHO, but instead, make this limit configurable in the server and have the service controllers validate it. For some deployments, a higher limit may be perfectly manageable and even desirable to reduce the pagination overhead.

Reply:
I think for now it is a good idea to have a hard limit at the API level; we can always raise it later